tests/virt_autotest/kubevirt_barriers.pm

```diff
@@ -0,0 +1,23 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Run kubevirt test suite
+#
+# Maintainer: Nan Zhang <nan.zhang@suse.com>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+
+sub run {
+    barrier_create('kubevirt_test_setup', 2);
+    barrier_create('rke2_server_start_ready', 2);
+    barrier_create('rke2_server_restart_complete', 2);
+    barrier_create('kubevirt_test_done', 2);
```

Comment:
Would you please explain what barrier_create do? Seems a little weird to have different barrier created together, which seem to have different pre-conditions.

Reply:
Besides, I just notice this test file is called for both server and agent , before the server and agent specific test setup start. I wonder what each of above barrier stands for ...

Reply:
I tried to call barrier_create on server only, but it doesn't work. If call it on both nodes, it works fine. Not sure if there is any other usage.

Reply:
@tonyyuan1  I remember you used to use barrier_create. Would you please give a review here if the usage is correct and suitable?

Reply:
@kalikiana  Hello Cris, I guess you should know this barrier_create api well? Would you please give a review if the usage here is correct and suitable? Or can you help to involve someone who is familiar? Thanks!

Reply:
I have no knowledge of kubevirt but I am wondering why it need so many barriers. A barrier can be used by multiple jobs.  Could some of them be reused?

Reply:
> I have no knowledge of kubevirt but I am wondering why it need so many barriers. A barrier can be used by multiple jobs. Could some of them be reused?

For kubevirt server&agent deployment, there are many setup points, it should be marked with different barrier names.

Reply:
@nanzhg  From https://github.com/os-autoinst/openQA/blob/master/docs/WritingTests.asciidoc#test-synchronization-and-locking-api, it is given below:

````
Mutexes are a way to wait for specific events from a single job. When we need multiple jobs to reach a certain state we need to use barriers.

To create a barrier call barrier_create with the parameters name and count. The name serves as an ID (same as with mutexes). The count parameter specifies the number of jobs needed to call barrier_wait to unlock barrier.
```

IMHO, you have 2 jobs in your job cluster, so your usage is correct, although not that necessary especially for the middle 2 locks.  

A note to you, that document section also has below info about option check_dead_job which can be useful for you.

```
An example would be one master and three worker jobs and you want to do initial setup in the three worker jobs before starting main actions. In such a case you might use check_dead_job to avoid useless actions when one of the worker jobs dies.
```

Reply:
I've figured out the synchronization issue. As tested, the job runs very stable.